src/Module.h

```diff
@@ -112,6 +114,13 @@ EXPORT void compile_standalone_runtime(const std::string &object_filename, Targe
  */
 EXPORT Outputs compile_standalone_runtime(const Outputs &output_files, Target t);
 
+typedef std::function<Module(const std::string &, const Target &)> ModuleProducer;
+
+EXPORT void compile_multitarget(const std::string &fn_name, 
+                                const Outputs &output_files,
+                                const std::vector<Target> &targets, 
+                                ModuleProducer module_producer);
```

Comment:
Can this just take a vector of modules instead? That seems simpler than a vector of targets and a lambda.